scripts/tuple_tree_generator/tuple_tree_generator/templates/struct.h.tpl

```diff
@@ -56,7 +64,11 @@ struct /*= struct.fullname =*/
     /**- for field in struct.key_fields **/
     /*=- field.name =*/(/*= field.name =*/)/** if not loop.last **/, /** endif **/
     /**- endfor **/
-    /**- endif **/ {}
+    /**- endif **/ {
+      /**- if struct.inherits and not 'Kind' in struct.key_fields | map(attribute='name') -**/
+      Kind = AssociatedKind;
```

Comment:
Why don't you do this in the initialization list?

Reply:
2 reasons:
* Kind belongs to the base class, it cannot be initialized in the initialization list
* Putting the kind in the parent's initialization (e.g. `Type(AssociatedKind)`) would mean foregoing the default constructor.

Also, this replaces the same [manual implementation](https://github.com/revng/revng/blob/de033b50f20019a2ca1d996e0b845e3962688cbe/include/revng/Model/StructType.h#L38)